internal/release/scripts/prep-release.sh

```diff
@@ -44,10 +59,15 @@ update_and_validate_build() {
 
         # Validate the build number format
         if [[ "${build}" =~ ^[0-9]{7}$ ]]; then
-            # Write the valid build number to the file
-            echo "${build}" > "${build_file}"
-            echo "New build number is: ${build}"
-            return 0
+            if (( 10#$current_build < 10#$build )); then
+                # Write the valid build number to the file
+                echo "${build}" > "${build_file}"
+                echo "New build number is: ${build}"
+                return 0
+            else
+                echo "Build version hasn't changed or is less than current build version. Stopping." >&2
+                cleanup
```

Comment:
```suggestion
                echo "Build version hasn't changed or is less than current build version."
```

Sorry for misguiding you, this should have remained a loop.